### ISSUE TITLE
Creating bookmark on the server failed, but it didn't fail

### ISSUE THREAD

**Issue by TheDevFreak** [marcelklehr/floccus#85]
### Software versions

* Browser: FF60.0
* Nextcloud: 13.0.1
* Nextcloud Bookmarks app:  0.11.0
* Floccus: 2.0.6

### Steps to reproduce

1. Remove bookmark from server
2. Create bookmark on client
3. Press Sync

### Expected outcome

Sync successful message

### Actual outcome

A good few "Creating bookmark on the server failed" messages, despite bookmarks actually being added the the server

**Comment by TheDevFreak** [comment]
False alarm, it's due to a reverse proxy configuration. Temporary fix is disabling brute force protection

**Comment by mobamoba** [comment]
I'd like to reopen this as running Nextcloud behind an Nginx reverse proxy causes this problem. No clue why as every other aspect of Nextcloud works fine behind the reverse proxy. How could we debug this as it's not an option for me to run Nextcloud without Nginx and Floccus is the only app that doesn't work with it? Thanks.

**Comment by marcelklehr** [comment]
@mobamoba I take it you've tried disabling brute force protection? Mmh. It would be interesting to see the reply from the server. Could you debug floccus and inspect the HTTP request that tries to create the bookmark in the network panel of the debugger?

**Comment by mobamoba** [comment]
It just gives a bunch of 504 Gateway Timeouts. I'm actually accessing my Nextcloud instance at the time Floccus is saying the gateway is timing out. The url it's using is correct - https://my.nextcloud.com:nginxport/index.php/apps/bookmarks/public/rest/v2/bookmark

Same thing happens in Chrome if that helps. When I switch it to the non-Nginx connection, it all syncs fine.

**Comment by marcelklehr** [comment]
Then it might be because the bookmarks app fetches the web pages you create bookmarks for. Can you increase the timeout in nginx?

**Comment by mobamoba** [comment]
Okay after a bunch of trial and error, here's what works for me at least with Floccus and Nextcloud - I'm don't really know enough about Nginx to know which part of this is the necessary part for Floccus but all is working now so I'm posting in case someone else has this issue:

> `location /{
	proxy_pass http://NEXTCLOUD/;
    proxy_buffering off;
    proxy_http_version 1.1;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $http_connection;
    proxy_request_buffering off;
    client_max_body_size 0;
    proxy_read_timeout  36000s;
    proxy_ssl_session_reuse off;          
            
   }`

**Comment by bfayers** [comment]
When I resolved this for myself I did this https://github.com/nextcloud/docker/issues/294#issuecomment-381418214 but im running through a reverse proxy, all this did was ensured nextcloud new of the correct IP which in turn made disabling bruteforce protection not be needed. So i know have functioning IPs inside the container and bruteforce protection on

**Comment by bhlevca** [comment]
One year later and I have the same problem. With nextcloud behind haproxy floccus cannot connect.
Everything else works fine. I tried all the suggestion form above . I don't have the brute force app , but I use fail2ban instead.  
My nextcloud server runs in virtual domain of an apache2 server and it is difficult for me to connect directly to that apache virtual domain using the IP.

Is there any way to debug this i looked in haproxy.logs and in the nextcloud logsand I found nothing relevant.

Thanks
